### PR TITLE
SW-1979 Move jOOQ configuration to a subproject

### PR DESCRIPTION
**Pull request by sgrimm**
Previously, our customization and configuration of jOOQ code generation (which
needs to be executed by Gradle as part of the build process, not by our
application code at runtime) lived in the `buildSrc` directory.

`buildSrc` has a problem: whenever it changes, the entire project is rebuilt.
Changing the jOOQ configuration would cause the MJML email templates to be
rebuilt even though they have no dependency relationship.

Move the jOOQ configuration to a separate subproject that produces a Gradle
plugin. Changes to the configuration will cause the plugin to be rebuilt, which
will trigger a fresh round of jOOQ code generation, but won't invalidate
unrelated parts of the project. Updating `Config.kt` won't cause MJML templates
to be rebuilt.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #708** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/708" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/708?utm_source=stack-comment).

File: jooq/src/main/kotlin/com/terraformation/backend/jooq/IdWrapper.kt [hunk 1]
```diff
@@ -3,10 +3,7 @@ package com.terraformation.backend.jooq
 import org.jooq.codegen.JavaWriter
 import org.jooq.meta.jaxb.ForcedType
 
-class IdWrapper(
-  private val className: String,
-  includeExpressions: List<String>
-) {
+class IdWrapper(private val className: String, includeExpressions: List<String>) {
```
Comment (sgrimm) [inline jooq/src/main/kotlin/com/terraformation/backend/jooq/IdWrapper.kt:6]: No actual changes to this file, just ktfmt reformatting.

File: jooq/src/main/kotlin/com/terraformation/backend/jooq/TerrawareGenerator.kt [hunk 2]
```diff
@@ -33,7 +33,8 @@ class TerrawareGenerator : KotlinGenerator() {
       import com.terraformation.backend.db.EnumFromReferenceTable
       import org.jooq.impl.AbstractConverter
 
-    """.trimIndent())
+    """
+            .trimIndent())
```
Comment (charlie-tf) [inline jooq/src/main/kotlin/com/terraformation/backend/jooq/TerrawareGenerator.kt:37]: what's the reason for these extra spaces?

Comment (charlie-tf) [reply jooq/src/main/kotlin/com/terraformation/backend/jooq/TerrawareGenerator.kt:37]: is just from ktfmt?

**Review by charlie-tf — APPROVED** [review]
👍

File: jooq/src/main/kotlin/com/terraformation/backend/jooq/TerrawareGenerator.kt (hunk 2 above)
Comment (sgrimm) [reply jooq/src/main/kotlin/com/terraformation/backend/jooq/TerrawareGenerator.kt:37]: Yeah, this is ktfmt. Its handling of multiline strings leaves something to be desired.

**Pull request merged**